src/main/drivers/timer.h

```diff
@@ -152,6 +152,8 @@ typedef enum {
     TIMER_OUTPUT_N_CHANNEL = (1 << 1),
 } timerFlag_e;
 
+#define TIM_N(n) (1 << (n))
```

Comment:
Why was this moved? Since this is nothing but `BIT()` by another name, I would prefer if we didn't extend it's scope, and instead tried to replace it with `BIT()`.

Reply:
You prefer replacing all occurrences of `TIM_N` to `BIT` now?

Or, some hybrid approach such as this?
- Replace the definition of `TIM_N` to `BIT` (`#define TIM_N(n) BIT(n)`).
- Current users of `TIM_N` continue to use `TIM_N` (`timer.c`, timer_hal.c` and all `target.h`)
- Use `BIT` elsewhere.

Reply:
On the contrary, I propose that we extend the scope of `TIM_N`, as `TIM_N` is not just a mere alias for `BIT` but it means `a bit that represents a timer`. (Replacing the definition of `TIM_N` to `BIT` is fine.)

Reply:
I am not a fan of introducing duplicity, and making it possible for contributors to use `TIM_N` and `BIT` interchangably - sooner or later we will end up with both of them being used in the same file, as arguments to the same function call, which I think we want to avoid.
My suggestion would have been to leave `TIM_N` defined in `timer.c` as it is, to keep the scope restricted, and then eliminate it completely once the custom target definitions are gone with 4.1. Eliminating it in target definitions now will introduce a lot of change that will lead to conflicts when cherry-picking target changes into the maintenance branch, which should be avoided if possible.

Reply:
Well, IMO, it's not a duplicity but adding a semantics to bit position.

But no fuss here. `TIM_N` is back into `timer.c` now.

Reply:
Agreed that it is adding semantics. But I do not know if in a statement like

```
#define USED_TIMERS  ( BIT(2) [...]
```

extra semantics are needed to make it clear that '2' is a timer number.

I also think that if we want to keep these extra semantics (and potentially introduce more similar ones in other places where `BIT()` is used to indicate an element in a list of numbered options), then keeping the scope restricted to a compilation unit (which is working out in a weird but justifiable way in the case of `timer.c`) is a better approach.